test/benchmark.py

```diff
@@ -0,0 +1,327 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A test suite to check Spanner dialect for SQLAlchemy performance
+in comparison with the original Spanner client.
+"""
+import datetime
+import random
+from scipy.stats import sem
+import statistics
+import time
+
+from google.api_core.exceptions import Aborted
+from google.cloud import spanner_dbapi
+from google.cloud.spanner_v1 import Client, KeySet
+from sqlalchemy import (
+    create_engine,
+    insert,
+    select,
+    text,
+    MetaData,
+    Table,
+)
+
+
+def measure_execution_time(function):
+    """Decorator to measure a wrapped method execution time."""
+
+    def wrapper(self, measures):
+        """Execute the wrapped method and measure its execution time.
+
+        Args:
+            measures (dict): Test cases and their execution time.
+        """
+        t_start = time.time()
+        try:
+            function(self)
+            measures[function.__name__] = round(time.time() - t_start, 2)
+        except Aborted:
+            measures[function.__name__] = 0
+
+    return wrapper
+
+
+class BenchmarkTestBase:
+    """Base class for performance testing.
+
+    Organizes testing data preparation and cleanup.
+    """
+
+    def __init__(self):
+        self._create_table()
+
+        self._one_row = (
+            1,
+            "Pete",
+            "Allison",
+            datetime.datetime(1998, 10, 6).strftime("%Y-%m-%d"),
+            b"123",
+        )
+
+    def _cleanup(self):
+        """Drop the test table."""
+        conn = spanner_dbapi.connect("sqlalchemy-dialect-test", "compliance-test")
+        conn.database.update_ddl(["DROP TABLE Singers"])
+        conn.close()
+
+    def _create_table(self):
+        """Create a table for performace testing."""
+        conn = spanner_dbapi.connect("sqlalchemy-dialect-test", "compliance-test")
+        conn.database.update_ddl(
+            [
+                """
+CREATE TABLE Singers (
+    id INT64,
+    first_name STRING(1024),
+    last_name STRING(1024),
+    birth_date DATE,
+    picture BYTES(1024),
+) PRIMARY KEY (id)
+        """
+            ]
+        ).result(120)
+
+        conn.close()
+
+    def run(self):
+        """Execute every test case."""
+        measures = {}
+        for method in (
+            self.insert_one_row_with_fetch_after,
+            self.read_one_row,
+            self.insert_many_rows,
+            self.select_many_rows,
+            self.insert_many_rows_with_mutations,
+        ):
+            method(measures)
+
+        self._cleanup()
+        return measures
+
+
+class SpannerBenchmarkTest(BenchmarkTestBase):
+    """The original Spanner performace testing class."""
+
+    def __init__(self):
+        super().__init__()
+        self._client = Client()
+        self._instance = self._client.instance("sqlalchemy-dialect-test")
+        self._database = self._instance.database("compliance-test2")
```

Comment:
The base class is creating the `compliance-test` database. From what I can tell `compliance-test2` isn't referenced anywhere else.
```suggestion
        self._database = self._instance.database("compliance-test")
```

Reply:
Oh. Locally I run tests on `compliance_test2` - it comes from times when we were using only one instance. This line slipped through my radars.